### PR TITLE
Provide a Coroutine variant of ScanStream/ScanIterator

### PR DESCRIPTION
**Pull request by sokomishalov**
Implements https://github.com/lettuce-io/lettuce-core/issues/1435

P.S. Actually I'm not sure about the extension's signature due to the compiler conflict with existing member methods.
It makes end-users provide import aliases (see `ScanStreamCoroutinesIntegrationTests`).
Also maybe we should not underly on the `ScanStream` and provide separate and independent one.

### CODE REVIEW

**Comment by mp911de** [comment]
Moving this into the next iteration. Will review this PR after the release of 6.0.

File: src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import kotlinx.coroutines.reactive.asFlow
+
+/**
+ * Sequentially iterate the keys space.
+ *
+ * @param scanArgs scan arguments.
+ * @return `Flow<K>` flow of keys.
+ */
+@JvmOverloads
+fun <K : Any, V : Any> RedisKeyCoroutinesCommands<K, V>.scan(scanArgs: ScanArgs? = null): Flow<K> {
```
Comment (mp911de) [inline src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt:16]: Overloading the keys commands for Coroutines uses a different design than the other API (sync, reactive) where `ScanStream`/`ScanIterator` are separate and the API package doesn't depend on `ScanStream`/`ScanIterator`.

Let's keep it the same style and add only an extension for `ScanStream` taking `RedisKeyCoroutinesCommands` as an argument and returning `Flow`.

File: src/test/kotlin/io/lettuce/core/api/coroutines/CoroutinesIntegrationTests.kt [hunk 2]
```diff
@@ -36,51 +37,39 @@ import javax.inject.Inject
  * @author Mark Paluch
  */
 @ExtendWith(LettuceExtension::class)
+@ExperimentalCoroutinesApi
```
Comment (mp911de) [inline src/test/kotlin/io/lettuce/core/api/coroutines/CoroutinesIntegrationTests.kt:40]: Let's move changes to this file into a separate commit.

**Review by mp911de — CHANGES_REQUESTED** [review]
I added two comments, one about the overall design, one about splitting this PR into two commits to keep commits focussed on the actual topic.

File: src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt (hunk 1 above)
Comment (sokomishalov) [reply src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt]: Adding extensions for `ScanStream` is useless and impossible in the right way for now.
https://youtrack.jetbrains.com/issue/KT-11968.
I propose to introduce `ScanFlow` singleton to access this API.

File: src/test/kotlin/io/lettuce/core/api/coroutines/CoroutinesIntegrationTests.kt (hunk 2 above)
Comment (sokomishalov) [reply src/test/kotlin/io/lettuce/core/api/coroutines/CoroutinesIntegrationTests.kt]: Agree, will move to a separate PR with kotlin tests pollishing

File: src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt (hunk 1 above)
Comment (mp911de) [reply src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt]: Works for me, too. Feel free to do it that way.

Comment (sokomishalov) [reply src/main/kotlin/io/lettuce/core/api/coroutines/RedisKeyCoroutinesCommandsExtensions.kt]: Done

File: src/test/kotlin/io/lettuce/core/ScanFlowIntegrationTests.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package io.lettuce.core
```
Comment (dengliming) [inline src/test/kotlin/io/lettuce/core/ScanFlowIntegrationTests.kt:16]: missing license header？

Comment (sokomishalov) [reply src/test/kotlin/io/lettuce/core/ScanFlowIntegrationTests.kt:16]: Yep, my bad. Added.

**Comment by mp911de** [comment]
I'm going to take this PR from here.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.
